packet.go

```diff
@@ -314,5 +466,110 @@ func (p *Packet) decodeIp6(recur int) {
 		p.decodeIcmp()
 	case IP_INIP:
 		p.decodeIp6(recur + 1)
+	case IP_GRE:
+		p.decodeGre()
+	}
+}
+func (p *Packet) decodeVxLAN() bool {
+	// VxLAN decode is a heuristic and not an exact algorithm.
+
+	// VXLAN Header is 8 bytes:
+	//  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+	// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	// |R|R|R|R|I|R|R|R|            Reserved                           |
+	// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	// |                VXLAN Network Identifier (VNI) |   Reserved    |
+	// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	if len(p.Payload) < 8 {
+		return false
+	}
+	pkt := p.Payload
+
+	// if 'I' bit is not set, there is no valid VNI.
+	// Which means it is very less likely to be a vxlan
+	// packet. We can safely return.
+	if uint8(pkt[0])&0x08 == 0 {
+		return false
+	}
+
+	// Seems like a possible vxlan payload.  We are not interested in vxlan
+	// header itself.  We want to know if inner ethernet frame and inner
+	// payloads are valid. Skip vxlan header.
+	p.Payload = p.Payload[8:]
```

Comment:
you need to undo setting p.Pyaload after this line, if you are returning false.

Reply:
thanks for catching it. I'm fixing it,